pkg/solana/logpoller/log_poller.go

```diff
@@ -525,3 +531,35 @@ func (lp *Service) backgroundWorkerRun(ctx context.Context) {
 func (lp *Service) FilteredLogs(ctx context.Context, queryFilter []query.Expression, limitAndSort query.LimitAndSort, queryName string) ([]Log, error) {
 	return lp.orm.FilteredLogs(ctx, queryFilter, limitAndSort, queryName)
 }
+
+func (lp *Service) computeLookbackWindow(ctx context.Context) (int64, error) {
+	latestFinalizedSlot, err := lp.client.SlotHeightWithCommitment(ctx, rpc.CommitmentFinalized)
+	if err != nil {
+		return 0, fmt.Errorf("error getting latest slot from RPC: %w", err)
+	}
+
+	if latestFinalizedSlot == 0 {
+		return 0, fmt.Errorf("latest finalized slot is 0 - waiting for next slot to start processing")
+	}
+
+	lookback := lp.startingLookback
+
+	// nolint:gosec
+	// G115: integer overflow conversion uint64 -&gt; int64
+	latestSlot := int64(latestFinalizedSlot)
+	lookbackSlot := latestSlot - int64((lookback-1)/lp.blockTime) - 1 // = latestSlot - ceil(lookback/blockTime)
+
+	firstAvailableSlot, err := lp.client.GetFirstAvailableBlock(ctx) // Despite the name, this returns slot # not block number
+	if err != nil {
+		// This is an optimization to avoid requesting pruned blocks. If there's an err we can just ignore
+		return lookbackSlot, nil
```

Comment:
In this function, we should log the final value we chose as the lookbackSlot, and why. This can be used for troubleshooting later, and to understand what/why did the node choose for LogPoller on a restart.